packages/webapp/src/containers/Map/LocationSelectionModal/index.jsx

```diff
@@ -5,16 +5,24 @@ import React from 'react';
 import { useSelector } from 'react-redux';
 import { canShowSelectionSelector, mapLocationsSelector } from '../../mapSlice';
 import PurePreviewPopup from '../../../components/Map/PreviewPopup';
+import { SENSOR } from '../../SensorReadings/constants';
 
-export default function LocationSelectionModal({ history, selectingOnly }) {
+export default function LocationSelectionModal({ history, sensorReadings, selectingOnly }) {
   const { dismissSelectionModal } = useSelectionHandler();
   const showSelection = useSelector(canShowSelectionSelector);
   const locations = useSelector(mapLocationsSelector);
-  if (showSelection && locations.length === 1 && locations[0].type === 'sensor') {
+  const locationSensorReadings = Object.values(sensorReadings).filter(
+    (reading) => reading.location_id === locations[0]?.id,
+  );
+  if (showSelection && locations.length === 1 && locations[0].type === SENSOR) {
     return (
       <div className={styles.selectionModal} onClick={dismissSelectionModal}>
         <div className={styles.selectionContainer}>
-          <PurePreviewPopup location={locations[0]} history={history} />
+          <PurePreviewPopup
```

Comment:
I'm not sure what the practice for this would be, but would it make more sense to create a container PurePreviewPopup, create a selector for getting readings by location_id from the redux store, and have the container retrieve the data from the store rather than passing the sensor readings from component to component like this. I feel that would be cleaner, but I'll leave it up to you.

Reply:
I updated so that LocationSelectionModal will retrieve readings by location_id and then pass them into the popup components.